CriteoPublisherSdk/Sources/Util/CR_URLResolver.m

```diff
@@ -66,6 +73,21 @@ - (void)resolverURL:(NSURL *)url
   [task resume];
 }
 
++ (BOOL)isAppStoreURL:(NSURL *)URL {
```

Comment:
There is only a `hasSuffix` and `hasPrefix` but honestly this is already a mental gym to just see that you're looking for URL like: `^(apps|itunes|books|music)\..*\.apple\.com$`.

Regex are difficult to read when they are complex. But in this case, I personally think that the regex is simpler than a combination of prefix/suffix checks.

Reply:
Some people would argue that regexes are evil in all cases, I won't, I think this is mostly a matter of personal taste.
One valid argument would be that some day, adding some rule, could easily defeat the regex being simple.
In any cases, I do not think this is worth the time rewriting this piece of code.